proposals/dao/tip_119.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  DeployUpgradeFunc,
+  NamedAddresses,
+  PcvStats,
+  SetupUpgradeFunc,
+  TeardownUpgradeFunc,
+  ValidateUpgradeFunc
+} from '@custom-types/types';
+import { BigNumber } from 'ethers';
+
+const toBN = BigNumber.from;
+
+/*
+
+Tribal Council Proposal: TIP-119: Add gOHM to Collaterisation Oracle
+
+1. Deploy gOHM oracle
+2. Set oracle on collaterisation oracle
+3. Add gOHM to CR 
+
+*/
+
+const fipNumber = 'tip_119';
+let pcvStatsBefore: PcvStats;
+
+// Do any deployments
+// This should exclusively include new contract deployments
+const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  ////////////// 1. Create and deploy gOHM USD oracle
+  const GOhmEthOracleFactory = await ethers.getContractFactory('GOhmEthOracle');
+  const gOhmEthOracle = await GOhmEthOracleFactory.deploy(addresses.core, addresses.chainlinkOHMV2EthOracle);
+  await gOhmEthOracle.deployed();
+
+  logging && console.log(`Deployed gOHM Eth Oracle at ${gOhmEthOracle.address}`);
+
+  // Create the gOHM USD oracle
+  const CompositeOracleFactory = await ethers.getContractFactory('CompositeOracle');
+  const gOhmUSDOracle = await CompositeOracleFactory.deploy(
+    addresses.core,
+    gOhmEthOracle.address,
+    addresses.chainlinkEthUsdOracleWrapper,
+    false
+  );
+
+  logging && console.log('Deployed gOHM oracle to: ', gOhmUSDOracle.address);
+  return {
+    gOhmUSDOracle,
+    gOhmEthOracle
+  };
+};
+
+// Do any setup necessary for running the test.
+// This could include setting up Hardhat to impersonate accounts,
+// ensuring contracts have a specific state, etc.
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  pcvStatsBefore = await contracts.collateralizationOracle.pcvStats();
+};
+
+// Tears down any changes made in setup() that need to be
+// cleaned up before doing any validation checks.
+const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in teardown for fip${fipNumber}`);
+};
+
+// Run any validations required on the fip using mocha or console logging
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  const gOhmEthOracle = contracts.gOhmEthOracle;
+  const gOhmUSDOracle = contracts.gOhmUSDOracle;
+  const collateralizationOracle = contracts.collateralizationOracle;
+  const gOHMHoldingPCVDeposit = contracts.gOHMHoldingPCVDeposit;
+
+  // 1. Validate gOHM ETH oracle price is valid
+  const gOhmETHPrice = (await gOhmEthOracle.read())[0];
+  // Eth price is ~$1000. gOHM price is ~$2400
+  // Therefore, gOHM price in ETH should be ~ (2400/1000) = 2.4 ETH
+  expect(toBN(gOhmETHPrice.value)).to.be.bignumber.at.least(ethers.constants.WeiPerEther.mul(1));
+  expect(toBN(gOhmETHPrice.value)).to.be.bignumber.at.most(ethers.constants.WeiPerEther.mul(5));
```

Comment:
it's fine to do this check here because proposal scripts are bounded in time, unlike integration tests (in a few weeks we won't be running this file anymore)